src/task/task.rs

```diff
@@ -235,15 +238,30 @@ impl Task {
 
     /// Get the named user defined attribute (UDA) in a legacy format.  This will return None for
     /// any key defined in the Task data model, regardless of whether it is set or not.
+    #[deprecated(note = "please use Task::get_user_defined_attribute")]
     pub fn get_legacy_uda(&self, key: &str) -> Option<&str> {
+        self.get_user_defined_attribute(key)
+    }
+
+    /// Get the named user defined attribute (UDA) This will return None for any key
+    /// defined in the Task data model, regardless of whether it is set or not.
+    pub fn get_user_defined_attribute(&self, key: &str) -> Option<&str> {
         if Task::is_known_key(key) {
             return None;
         }
         self.data.get(key)
     }
 
     /// Like `get_udas`, but returning each UDA key as a single string.
+    #[deprecated(
+        note = "please use Task::get_user_defined_attributes, legacy UDAs will be first class citizens in the future"
```

Comment:
```suggestion
        note = "please use Task::get_user_defined_attributes"
```
I don't think we need to offer a reason here (and similarly for other places this appears)